qutebrowser/browser/network/networkmanager.py

```diff
@@ -358,6 +358,23 @@ def createRequest(self, op, req, outgoing_data):
             dnt = '0'.encode('ascii')
         req.setRawHeader('DNT'.encode('ascii'), dnt)
         req.setRawHeader('X-Do-Not-Track'.encode('ascii'), dnt)
+
+        def same_domain():
+            tabbed_browser = objreg.get('tabbed-browser', scope='window', window=self._win_id)
+            view = tabbed_browser.currentWidget()
+            print(req.url().host() == view.url().host(), repr(view.url().host()), repr(req.url().host()))
+            # TODO: We probably want to allow headers if we're on test.com and
+            # doing a request to www.test.com? Or maybe a new settings value for
+            # this?
+            return req.url().host() == view.url().host()
```

Comment:
I don't like vendoring libraries either way, as it creates problems with packaging for distributions like Debian/Fedora which is desirable for the future.

We also can't use its downloading feature anyways, as it blocks which is undesirable in qutebrowser. I'll look at adding something to qutebrowser's `urlutils` for it as it doesn't seem to hard, but it might take me some days as I have some other things I started already.